src/lib/support/ScopedBuffer.h

```diff
@@ -150,14 +150,41 @@ class ScopedMemoryBuffer : public Impl::ScopedMemoryBufferBase<MemoryManagement>
     ScopedMemoryBuffer & Calloc(size_t elementCount)
     {
         Base::Calloc(elementCount, sizeof(T));
+        ExecuteConstructors(elementCount);
         return *this;
     }
 
     ScopedMemoryBuffer & Alloc(size_t elementCount)
     {
         Base::Alloc(elementCount * sizeof(T));
+        ExecuteConstructors(elementCount);
         return *this;
     }
+
+private:
+    template <typename U = T, std::enable_if_t<std::is_trivial<U>::value, int> = 0>
+    void ExecuteConstructors(size_t elementCount)
+    {
+        // Do nothing if our type is trivial.  In particular, if we are a buffer
+        // of integers, we should not go zero-initializing them here: either
+        // caller wants that and called Calloc(), or it doesn't and we shouldn't
+        // do it.
+    }
+
+    template <typename U = T, std::enable_if_t<!std::is_trivial<U>::value, int> = 0>
+    void ExecuteConstructors(size_t elementCount)
```

Comment:
What about the destructors though... This seems very risky re: expectations of contructor/destructor...

A ScopedMemoryBuffer should have only allowed bytes, really, since otherwise we are half-rebuilding std::vector/std::unique_ptr in a weird way

Reply:
> What about the destructors though... 

That's why there's still a static assert that the _destructor_ is trivial.

> otherwise we are half-rebuilding std::vector/std::unique_ptr in a weird way

We sort of are, yes, using our own allocator...